### ISSUE TITLE
UnpicklingError: Attempting to unpickle unsafe module collections from Java client

### ISSUE THREAD

**Issue by HackRanger** [graphite-project/graphite-web#1499]
Hello,

I am getting to setup graphite on Ubuntu 14.04. I am installing the graphite-web deb from repo. 
If I send the metrics from bash nc , I could see the graphs. But when I send the metrics from a java client, I get this exception from graphite. 

`Fri Apr 08 13:21:48 2016 :: Exception Caught
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/graphite/render/datalib.py", line 231, in fetchData
    cachedResults = CarbonLink.query(dbFile.real_metric)
  File "/usr/lib/python2.7/dist-packages/graphite/render/datalib.py", line 140, in query
    results = self.send_request(request)
  File "/usr/lib/python2.7/dist-packages/graphite/render/datalib.py", line 166, in send_request
    result = self.recv_response(conn)
  File "/usr/lib/python2.7/dist-packages/graphite/render/datalib.py", line 181, in recv_response
    return unpickle.loads(body)
  File "/usr/lib/python2.7/dist-packages/graphite/util.py", line 108, in loads
    return pickle_obj.load()
  File "/usr/lib/python2.7/dist-packages/graphite/util.py", line 97, in find_class
    raise pickle.UnpicklingError('Attempting to unpickle unsafe module %s' % module)
UnpicklingError: Attempting to unpickle unsafe module collections`

Here is my java code: 

`for (int i = 1; i < 1001; i++) {
                Socket conn = new Socket("carbon.graphiteserver.com", 2003);
                DataOutputStream dos = new DataOutputStream(conn.getOutputStream());
                Date d = new Date();
                d.setTime(start.toEpochSecond(ZoneOffset.UTC));

```
            dos.writeBytes("autoscale.user.arrival " + i + " " + d.getTime() + "\n");
            System.out.printf("%s",
                    "autoscale.user.arrival " + i + " " + start.toEpochSecond(ZoneOffset.UTC) + "\n");
            conn.close();
            start.plusMinutes(i + 10);
            Thread.sleep(10);
```

}`

Could anyone please help me out on this.

Thanks!

**Comment by deniszh** [comment]
Could you please advice which Java client you using? And which Graphite version?

Thanks!

**Comment by HackRanger** [comment]
Hi,
Graphite package version in 0.9.12+debian-3.
`$graphite-manage --version`
`1.6.1`
The java client was written by me to populate some data.

**Comment by deniszh** [comment]
Graphite 0.9.12 is quite old, this issue (https://github.com/graphite-project/graphite-web/issues/608) was fixed in 0.9.13 and later.

**Comment by HackRanger** [comment]
Thanks for your suggestion. It was problem with the string formatting. After digging into carbon logs, it was telling me about invalid line. I had to type cast time stamp to long.
